src/platform_impl/linux/wayland/window/state.rs

```diff
@@ -840,6 +848,35 @@ impl WindowState {
         }
     }
 
+    /// Make window background blurred
+    #[inline]
+    pub fn set_blur(&mut self, blurred: bool) {
+        if blurred && !self.is_blurred {
+            if self.blur.is_none() {
+                self.blur = self
+                    .blur_manager
+                    .as_ref()
+                    .map(|kbm| kbm.blur(self.window.wl_surface(), &self.queue_handle));
+            }
+            if let Some(blur) = &self.blur {
+                blur.commit();
+            } else {
+                info!("Blur manager unavailable, unable to change blur")
+            }
+        } else if !blurred && self.is_blurred {
+            if let Some(blur_manager) = &self.blur_manager {
+                blur_manager.unset(self.window.wl_surface());
+            } else {
+                info!("Blur manager unavailable, unable to change blur")
+            }
+            if let Some(blur) = &self.blur {
```

Comment:
Could also move the `blue_manager` logic there and use `unwrap`, since it can't happen that you have `blur`, but not manager.

```suggestion
            if let Some(blur) = self.blur.take() {
```